webxrlayers-1.bs

```diff
@@ -912,63 +915,50 @@ WebXR Device API Integration {#webxrintegration}
-  readonly attribute FrozenArray&lt;XRGenericLayer&gt;? layers;
+  readonly attribute FrozenArray&lt;XRLayer&gt;? layers;
 };
 </pre>
 
-
 The <dfn attribute for="XRRenderState">layers</dfn> attribute returns an array containing
-the instances of {{XRLayer}} or {{XRWebGLLayer}} that are displayed by the [=XR Compositor=].
+the instances of {{XRLayer}} that are displayed by the [=XR Compositor=].
 
 <div class="algorithm" data-algorithm="initialize-renderstate-for-layers">
-When an {{XRRenderState}} object |state| is created for an {{XRSession}} |session| with "[=feature descriptor/layers=]" enabled, the user agent MUST <dfn dfn-for="layers">initialize the render state</dfn> by running the following steps:
-  1. Initialize |state| by running the steps to [=initialize the render state=].
-  1. Initialize |state|'s {{XRRenderState/layers}} to an empty array.
+This module replaces the steps given by [=initialize the render state=]. Instead when an {{XRRenderState}} object |state|
+is created for an {{XRSession}} |session|, the user agent MUST <dfn dfn-for="layers">initialize the render state</dfn> by running the following steps:
+  1. Initialize |state| by running the original steps to [=initialize the render state=].
+  1. Initialize |state|'s {{XRRenderState/layers}} as follows:
+    <dl class="switch">
+      <dt class="switch">If the |session| was created with "[=feature descriptor/layers=]" enabled
+      <dd> Initialize |state|'s {{XRRenderState/layers}} with a new empty array.
+      <dt> Otherwise
+      <dd> set |state|'s {{XRRenderState/layers}} to <code>null</code>.
+    </dl>
 
 </div>
 
 updateRenderState changes {#updaterenderstatechanges}
 -------------------------
-<div class="algorithm" data-algorithm="updateRenderState-for-layers">
-When the <dfn method for="XRSession" dfn-for="layers">updateRenderState(|newState|)</dfn> method is called on an {{XRSession}} |session|
-with "[=feature descriptor/layers=]" enabled, the user agent MUST run the following steps:
 
-  1. Run {{XRSession/updateRenderState}} with |newState|.
-  1. If |newState|'s [=XRRenderStateInit/baseLayer=] is set, set |session|'s [=pending render state=] to <code>null</code>, throw an {{InvalidStateError}} and abort these steps.
+<div class="algorithm" data-algorithm="update-layers-state-for-layers">
+
+This module replaces the steps given by "[=update the pending layers state=]". Instead when the user agent will
+<dfn dfn-for="layers">update the pending layers state</dfn> with {{XRSession}} |session| and {{XRRenderStateInit}} |newState|, it must run the following steps:
+  1. If the |session| was not created with "[=feature descriptor/layers=]" enabled:
+    1. If {{XRRenderStateInit/layers}} is not <code>null</code>, throw an {{NotSupportedError}} and abort these steps.
+    1. Abort these steps.
+  1. If |newState|'s {{XRRenderStateInit/baseLayer}} is set, throw a {{InvalidStateError}} and abort these steps.
```

Comment:
These steps are now called unconditionally, so this should check for both layers and baselayer being set

Reply:
Are you saying `layers` shouldn't be null?
Like: If baseLayers is set or layers is null, throw InvalidStateError

I think that'd be wrong.

Reply:
Are you saying layers shouldn't be null?
Like: If baseLayers is set or layers is null, throw InvalidStateError

I think that would be wrong.

Reply:
I don't think I understand you, what I'm saying is that the new "update the pending layers state" algorithm is unconditionally called, even when the feature is not set (this is good).

This should only error in cases when layers are being used -- either by checking the feature as discussed below, or when pending/active's layers is also set.

We should just give this algorithm two branches for when layers is enabled vs not enabled.

Reply:
That is what it is doing. Line 951 is true if layers isn't requested.

Reply:
Oh! I misread that, never mind!